File: ads/src/main/java/com/ads/module/funtion/RewardCallback.java
package com.ads.module.funtion;

import com.google.android.gms.ads.rewarded.RewardItem;

public interface RewardCallback {
    /** The vendor reported that the rewarded ad reached the screen. */
    default void onRewardedAdShown() {}

    /** The vendor reported an impression for the rewarded ad. */
    default void onAdImpression() {}

    void onUserEarnedReward(RewardItem var1);

    void onRewardedAdClosed();

    void onRewardedAdFailedToShow(int codeError);

    void onAdClicked();
}

File: ads/src/main/java/com/ads/module/funtion/RewardCallback.kt
package com.ads.module.funtion

import com.google.android.gms.ads.rewarded.RewardItem

interface RewardCallback {
    /** The vendor reported that the rewarded ad reached the screen. */
    fun onRewardedAdShown() {}

    /** The vendor reported an impression for the rewarded ad. */
    fun onAdImpression() {}

    fun onUserEarnedReward(var1: RewardItem?)

    fun onRewardedAdClosed()

    fun onRewardedAdFailedToShow(codeError: Int)

    fun onAdClicked()
}
